libcflib/db.xsh

```diff
@@ -0,0 +1,82 @@
+"""Database for accessing graph information"""
+import os
+
+import zict
+
+from .model import Artifact
+
+class DB:
+    """A logging object for fixie that stores information in line-oriented JSON
+    format.
+    """
+
+    __inst = None
+
+    def __new__(cls):
+        # make the logger a singleton
+        if DB.__inst is None:
+            DB.__inst = object.__new__(cls)
+        return DB.__inst
+
+    def __init__(self, history=100):
```

Comment:
I think history should be called something like `cache_size` and should default to 1 Million